dotCMS/src/main/java/com/dotcms/util/ThreadContextUtil.java

```diff
@@ -0,0 +1,132 @@
+package com.dotcms.util;
+
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.util.UtilMethods;
+
+/**
+ * Thread Context Util provides methods to handle reindex stuff and other thread local context things
+ * @author jsancas
+ */
+public class ThreadContextUtil {
+
+    private static ThreadLocal<ThreadContext> contextLocal = new ThreadLocal<>();
+    private static final ThreadContextUtil     INSTANCE    = new ThreadContextUtil();
+
+    public static ThreadContextUtil getInstance () {
+
+        return INSTANCE;
+    }
+    /**
+     * Get the context from the current thread
+     * @return {@link ThreadContext}
+     */
+    public ThreadContext getContext () {
+
+        return contextLocal.get();
+    }
+
+
+    /**
+     * Set the current thread context
+     */
+    public void setContext (final ThreadContext  context) {
```

Comment:
why have a setContext? ThreadContext is mutable (which is fine)

Reply:
Done